### ISSUE TITLE
BUG: order of all_obs_names is not consistent

### ISSUE THREAD

**Issue by iosonofabio** [snap-stanford/SATURN#83]
Hi there and thanks for the great package (we met a few times in Greece and San Diego).

I'm using SATURN with a lot of species and I noticed a pretty severe bug in the cell name assignments. The variable `all_obs_names` is the result of iterating over a Python dict, which is not guaranteed to follow alphabetical order of the keys (it does not on my machine). However, when these cell names are paired with the macrogene matrices (e.g. after pretrain and after metric learning), the matrix rows are ordered by pytorch's autobatcher over the dictionary, which uses alphabetical order (again, on my pytorch version, which is probably more recent than the one you developed the package on).

As a consequence, all the `obs_names` of the resulting `adata` objects are shuffled randomly.

I could make a quick PR fixing this by ensuring `all_obs_names` is alphabetical, or a slightly deeper one that changes the batch collate function to yield the cell names as well.

What do you think?

**Comment by Yanay1** [comment]
What version of python are you using? Can you try upgrading to 3.7+ https://stackoverflow.com/questions/5629023/order-of-keys-in-dictionaries-in-old-versions-of-python

**Comment by iosonofabio** [comment]
3.10-3.12

Can you please not close the issue? This is an actual bug, I fixed it in my fork but I had to add a sort statement to avoid it from crashing. Thanks

**Comment by cchd0001** [comment]
@iosonofabio  I also trigger this bug that the meta data in obs are not exactly match the input, could you please tell me how to fix this ?  My python is Python 3.10.12

**Comment by iosonofabio** [comment]
Use my fork 😜

This is definitely not a closed issue btw since it was not fixed
